pkg/agent/controller/networkpolicy/reconciler_test.go

```diff
@@ -703,3 +704,186 @@ func TestReconcilerUpdate(t *testing.T) {
 		})
 	}
 }
+
+func TestGroupPodsByServices(t *testing.T) {
+	port80 := intstr.FromInt(80)
+	port443 := intstr.FromInt(443)
+	port8080 := intstr.FromInt(8080)
+	portHTTP := intstr.FromString("http")
+	portHTTPS := intstr.FromString("https")
+	protocolTCP := v1beta1.ProtocolTCP
+	serviceTCP80 := v1beta1.Service{Protocol: &protocolTCP, Port: &port80}
+	serviceTCP8080 := v1beta1.Service{Protocol: &protocolTCP, Port: &port8080}
+	serviceTCP443 := v1beta1.Service{Protocol: &protocolTCP, Port: &port443}
+	serviceHTTP := v1beta1.Service{Protocol: &protocolTCP, Port: &portHTTP}
+	serviceHTTPS := v1beta1.Service{Protocol: &protocolTCP, Port: &portHTTPS}
```

Comment:
nit: we seem to be using some of these in multiple tests, maybe define top-level variables to reduce overall verbosity